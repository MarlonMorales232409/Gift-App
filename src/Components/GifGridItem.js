import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ title, url }) => {
  return (
    <div className="gift animate__animated animate__bounce animate__fadeIn">
      <img src={url} alt={title} className="gift-image" />
      <p className="gift-title">{title}</p>
    </div>
  );
};

export default GifGridItem;

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
