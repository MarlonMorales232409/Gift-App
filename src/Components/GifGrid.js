import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import Loader from "./Loader";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  const { loading, data: images } = useFetchGifs(category);

  return (
    <>
      <h2 className="category-h2 animate__animated animate__bounce animate__fadeIn">
        {category}
      </h2>
      {loading && <Loader />}
      <hr className="category-divisor" />
      <div className="gift-container">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} category={category} />
        ))}
      </div>
    </>
  );
};

export default GifGrid;
