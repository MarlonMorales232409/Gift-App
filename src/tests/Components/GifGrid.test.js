import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../Components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import GifGridItem from "../../Components/GifGridItem";
jest.mock("../../hooks/useFetchGifs");

describe("It shloud show it correctle", () => {
  let category = "vader";

  test("should show it correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show it correctly when data is coming from useFetch", () => {
    const gift = [
      {
        id: 123,
        url: "https://algo/algomas.jpg",
        title: "algo",
      },
      {
        id: 123,
        url: "https://algo/algomas.jpg",
        title: "algo",
      },
      {
        id: 123,
        url: "https://algo/algomas.jpg",
        title: "algo",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gift,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gift.length);
  });
});
