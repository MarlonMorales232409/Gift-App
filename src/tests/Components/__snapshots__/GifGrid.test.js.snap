// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It shloud show it correctle should show it correctly 1`] = `
<Fragment>
  <h2
    className="category-h2 animate__animated animate__bounce animate__fadeIn"
  >
    vader
  </h2>
  <Loader />
  <hr
    className="category-divisor"
  />
  <div
    className="gift-container"
  />
</Fragment>
`;

exports[`It shloud show it correctle should show it correctly when data is coming from useFetch 1`] = `
<Fragment>
  <h2
    className="category-h2 animate__animated animate__bounce animate__fadeIn"
  >
    vader
  </h2>
  <hr
    className="category-divisor"
  />
  <div
    className="gift-container"
  >
    <GifGridItem
      category="vader"
      id={123}
      key="123"
      title="algo"
      url="https://algo/algomas.jpg"
    />
    <GifGridItem
      category="vader"
      id={123}
      key="123"
      title="algo"
      url="https://algo/algomas.jpg"
    />
    <GifGridItem
      category="vader"
      id={123}
      key="123"
      title="algo"
      url="https://algo/algomas.jpg"
    />
  </div>
</Fragment>
`;
